// mssql/lib/table.js
// Generated by CoffeeScript 1.7.1
(function() {
  var Table,
    __slice = [].slice;

  Table = (function() {
    function Table() {
      this.columns = [];
      this.rows = [];
      Object.defineProperty(this.columns, "add", {
        value: function(name, column) {
          if (column instanceof Function) {
            column = column();
          }
          column.name = name;
          return this.push(column);
        }
      });
      Object.defineProperty(this.rows, "add", {
        value: function() {
          var values;
          values = 1 <= arguments.length ? __slice.call(arguments, 0) : [];
          return this.push(values);
        }
      });
    }

    Table.fromRecordset = function(recordset) {
      var col, name, row, t, _i, _len, _ref, _ref1;
      t = new this;
      _ref = recordset.columns;
      for (name in _ref) {
        col = _ref[name];
        t.columns.add(name, {
          type: col.type,
          length: col.length,
          scale: col.scale,
          precision: col.precision
        });
      }
      for (_i = 0, _len = recordset.length; _i < _len; _i++) {
        row = recordset[_i];
        (_ref1 = t.rows).add.apply(_ref1, (function() {
          var _j, _len1, _ref1, _results;
          _ref1 = t.columns;
          _results = [];
          for (_j = 0, _len1 = _ref1.length; _j < _len1; _j++) {
            col = _ref1[_j];
            _results.push(row[col.name]);
          }
          return _results;
        })());
      }
      return t;
    };

    return Table;

  })();

  module.exports = Table;

}).call(this);
